pytorch_lightning/utilities/types.py

```diff
@@ -0,0 +1,8 @@
+from typing import Dict, Iterator, Optional, Union
+
+import torch
+from torchmetrics import Metric
+
+_METRIC = Union[Metric, torch.Tensor, int, float]
+_STEP_OUTPUT = Optional[Union[torch.Tensor, Dict[str, torch.Tensor]]]
+_PARAMETERS = Iterator[torch.nn.Parameter]
```

Comment:
```suggestion
_HINT_METRIC = Union[Metric, torch.Tensor, int, float]
_HINT_STEP_OUTPUT = Optional[Union[torch.Tensor, Dict[str, torch.Tensor]]]
_HINT_PARAMETERS = Iterator[torch.nn.Parameter]
```
or keep typing just to keep clear from the first look that it typing Constants...

Reply:
Well, adding hint seems way too verbose

And keep in mind you do `from typing import Any`, not `from typing import AnyType`

Reply:
yes I see but how many times are you scrolling up to see whom where it is imported...
on the other hand we have some other func names too long and this is just not clear at the fist look
so lets stay with word TYPES in the var name?

Reply:
I personally prefer no suffixes. It should be obvious that it's a type considering its placement (`: METRIC` or `-> METRIC:`)

🎉 : `_TYPE` suffix (`_METRIC_TYPE`)
🚀 : No suffix (`_METRIC`)

cc: @PyTorchLightning/core-contributors for opinions as this will impact all codebase

Reply:
in the context of exposing some to the public like step types, it could be better to have the TYPE sufix